app/models/host_status/configuration_status.rb

```diff
@@ -27,12 +27,15 @@ def expected_report_interval
 
     def reported_origin_interval
       if last_report.origin
-        if host.params.has_key? "#{last_report.origin.downcase}_interval"
-          interval = host.params["#{last_report.origin.downcase}_interval"]
-        else
-          interval = Setting[:"#{last_report.origin.downcase}_interval"]
+        interval = nil
+        begin
+          if host.params.has_key? "#{last_report.origin.downcase}_interval"
+            interval = host.params["#{last_report.origin.downcase}_interval"]
+          end
+        rescue NameError, Safemode::SecurityError, Safemode::NoMethodError => exception
```

Comment:
I'm not sure I like catching these exceptions here in the model layer. A user doesn't have access to the logs so they don't know their parameter is silently ignored.

Implementation wise I'd really want it to be:
```ruby
def reported_origin_interval
  return unless last_report&.origin

  name = "#{last_report.origin.downcase}_interval"
  host.params.fetch(name) { Setting[name.to_sym] }
end
```

Looking at the code I think `host.params` will render all the parameters, so just calling that should be enough to trigger it an exception.

Thinking about solving it at a deeper level: should parameters be rendered before saving? Can that reliably work? It may still not be enough.

Perhaps `HostParams` needs a `safe_params` that ignores any invalid ERB. The reason I suggest this is that we use params in more places so I suspect there are more ways this bug could surface.